src/plugins/data/public/actions/value_click_action.ts

```diff
@@ -39,18 +43,32 @@ export function createValueClickActionDefinition(
     id: ACTION_VALUE_CLICK,
     shouldAutoExecute: async () => true,
     isCompatible: async (context: ValueClickContext) => {
+      if (context.data.query && isOfAggregateQueryType(context.data.query)) {
+        const queryString = await appendFilterToESQLQueryFromValueClickAction(context.data);
+        return queryString != null;
+      }
       const filters = await createFiltersFromValueClickAction(context.data);
       return filters.length > 0;
     },
     execute: async (context: ValueClickActionContext) => {
       try {
-        const filters: Filter[] = await createFiltersFromValueClickAction(context.data);
-        if (filters.length > 0) {
-          await getStartServices().uiActions.getTrigger(APPLY_FILTER_TRIGGER).exec({
-            filters,
+        if (context.data.query && isOfAggregateQueryType(context.data.query)) {
+          // ES|QL charts have a different way of applying filters,
+          // they are appending a where clause to the query
+          const queryString = appendFilterToESQLQueryFromValueClickAction(context.data);
+          await getStartServices().uiActions.getTrigger('UPDATE_ESQL_QUERY_TRIGGER').exec({
```

Comment:
Nit: Can we use a constant (and export it) for `'UPDATE_ESQL_QUERY_TRIGGER'`?

Reply:
The reason I didnt do it is because if I export it from the text-based languages plugin (where the trigger is being registered) it will create circular dependencies with the data plugin. I could export it from the esql-utils package but it seems a bit unrelated to me. So I will leave it as it is for now and I will think about it more when another usage will be required.